core/arch/x86/clean_call_opt.c

```diff
@@ -60,10 +60,10 @@ analyze_callee_regs_usage(dcontext_t *dcontext, callee_info_t *ci)
 {
     instrlist_t *ilist = ci->ilist;
     instr_t *instr;
-    uint i, num_regparm;
+    int i, num_regparm;
 
     ci->num_simd_used = 0;
-    memset(ci->simd_used, 0, sizeof(bool) * MCXT_NUM_SIMD_SLOTS);
+    memset(ci->simd_used, 0, sizeof(bool) * proc_num_simd_registers());
```

Comment:
I think the aarch64 comment would be good here as well.

Reply:
Yes, I think in AAarch64 it may not even happen at all actually, since AFAIK there is no feature bit that changes no. of SIMD registers dependent on the processor. Having said that, maybe I shouldn't even add the comment there and instead, add ASSERT(proc_num_simd_registers() == MCXT_NUM_SIMD_SLOTS)?

Reply:
OK.

Reply:
Done.